addons/payment_adyen/utils.py

```diff
@@ -53,15 +53,21 @@ def format_partner_address(partner):
     """
     STREET_FORMAT = '%(street_number)s/%(street_number2)s %(street_name)s'
     street_data = split_street_with_params(partner.street, STREET_FORMAT)
-    return {
+    stateOrProvince = partner.state_id.code
+    if partner.country_id.code not in ["CA", "US", "GB"]:
+        stateOrProvince = stateOrProvince or ''
+    address = {
         'city': partner.city,
         'country': partner.country_id.code or 'ZZ',  # 'ZZ' if the country is not known.
-        'stateOrProvince': partner.state_id.code,
+        'stateOrProvince': stateOrProvince,
         'postalCode': partner.zip,
         # Fill in the address fields if the format is supported, or fallback to the raw address.
         'street': street_data.get('street_name', partner.street),
         'houseNumberOrName': street_data.get('street_number'),
     }
+    if False in address.values():
+        raise ValueError("Please fill out address info")
```

Comment:
```suggestion
        raise ValueError("Please fill out the address state.")
```

Reply:
I don't agree, it's another check if other address values are not False.

Reply:
https://docs.adyen.com/risk-management/avs-checks/

Reply:
Ok but then you're adding a second check, is it intended?